shared/teams/team/member/add-to-channels.tsx

```diff
@@ -186,96 +189,224 @@ const AddToChannels = (props: Props) => {
+const HeaderRow = ({mode, onCreate, onSelectAll, onSelectNone}) => (
   <Kb.Box2
     direction="horizontal"
     alignItems="center"
     fullWidth={true}
-    style={Styles.collapseStyles([styles.item, styles.headerItem])}
+    style={Styles.collapseStyles([styles.item, styles.headerItem, mode === 'self' && styles.itemSelf])}
   >
     <Kb.Button label="Create channel" small={true} mode="Secondary" onClick={onCreate} />
-    <Kb.Text type="BodyPrimaryLink" onClick={onSelectAll || onSelectNone}>
-      {onSelectAll ? 'Select all' : 'Clear'}
-    </Kb.Text>
+    {mode === 'self' ? (
+      <Kb.Box />
+    ) : (
+      <Kb.Text type="BodyPrimaryLink" onClick={onSelectAll || onSelectNone}>
+        {onSelectAll ? 'Select all' : 'Clear'}
+      </Kb.Text>
+    )}
   </Kb.Box2>
 )
 
-const ChannelRow = ({channelname, conversationIDKey, numMembers, selected, onSelect}) => {
+const SelfChannelActions = ({meta}: {meta: ChatTypes.ConversationMeta}) => {
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const isAdmin = false
+  const inChannel = meta.membershipType === 'active'
+  const onEditChannel = () => {} // TODO: show another modal with a back button to here
+  const actionProps = {conversationIDKey: meta.conversationIDKey, teamID: meta.teamID}
+  const onAdministrate = () =>
+    dispatch(nav.safeNavigateAppendPayload({path: [{props: actionProps, selected: 'teamChannel'}]}))
+  const onDelete = () =>
+    // TODO: reload this list or at least hide this row
+    dispatch(nav.safeNavigateAppendPayload({path: [{props: actionProps, selected: 'teamDeleteChannel'}]}))
+  const onLeave = () =>
+    dispatch(
+      ChatGen.createLeaveConversation({conversationIDKey: meta.conversationIDKey, dontNavigateToInbox: true})
+    )
+  const onJoin = () => dispatch(ChatGen.createJoinConversation({conversationIDKey: meta.conversationIDKey}))
+
+  const menuItems = [
+    {onClick: onEditChannel, title: 'Edit channel'},
+    ...(isAdmin
+      ? [
+          {onClick: onAdministrate, title: 'Administrate'},
+          {danger: true, onClick: onDelete, title: 'Delete'},
+        ]
+      : []),
+  ]
+  const {popupAnchor, showingPopup, toggleShowingPopup, popup} = Kb.usePopup(getAttachmentRef => (
+    <Kb.FloatingMenu
+      attachTo={getAttachmentRef}
+      visible={showingPopup}
+      onHidden={toggleShowingPopup}
+      closeOnSelect={true}
+      items={menuItems}
+    />
+  ))
+  const [buttonMousedOver, setMouseover] = React.useState(false)
+  return (
+    <Kb.Box2 direction="horizontal" gap="xtiny" fullHeight={true}>
+      {popup}
+      {meta.channelname !== 'general' && (
+        // TODO: waiting state
+        // TODO: flip this to the other button after being clicked
+        <Kb.Button
+          type={buttonMousedOver && inChannel ? 'Default' : 'Success'}
+          mode={inChannel ? 'Secondary' : 'Primary'}
+          label={!inChannel ? 'Join' : buttonMousedOver ? 'Leave' : 'In'}
```

Comment:
prefer to remove nested ternary, or at least test the positive for both of them. Also, it's unclear what a button labeled "In" does on mobile.

Reply:
I rewrote the ternary to be positive for both. I think it's more readable now and more readable than it would be if switched to an if thing elsewhere, but I could be convinced otherwise.

Re "In": @cecileboucheron what do you think? I agree with Danny that a button labelled "in" isn't terribly clear that what it does is "leave channel"

Reply:
Sorry just seeing this. Let's keep it read "In" for now, does it have the green checkmark next to it like on the mockup?

Reply:
It does